### DISCUSSION TITLE
Authentication Error

### DISCUSSION THREAD

**Discussion by makaloni6** [saleweaver/python-amazon-sp-api#150] [Q&A] [answered]
I'm new to the SP api and I'm having trouble authenticating my calls.
Having set up a developer account, walked through the application registration process in Amazon's documentation.
Authentication does not work.

Here is my code

```
from sp_api.api import Orders
from sp_api.api import Reports
from sp_api.api import Feeds
from datetime import datetime
from datetime import timedelta

credentials= dict(
refresh_token='XXXXXXXX',
lwa_client_secret='XXXXXXXX',
aws_secret_key='XXXXXXXX',
aws_access_key='XXXXXXXX',
role_arn='XXXXXXXX',
)

res = Orders(credentials=credentials).get_orders(CreatedAfter=(datetime.now() - timedelta(days=7)).isoformat())
print(res.payload) 
```
I'm receiving the following error: 
  File "/Users/XXX/opt/anaconda3/lib/python3.7/site-packages/cachetools/cache.py", line 44, in __getitem__
    return self.__data[key]
KeyError: 'access_token_XXXXXX'

I've grabbed each of that params for their respective places. 
I've checked many times if these are correct.

I didn't set parameter access_token.
Why does this error occurred? and How do I solve it?

Any help would be greatly appreciated.

**Comment by saleweaver** [comment, marked as answer]
Either the credentials or your IAM setup are invalid. Please check all the other topics for more information (the exact same question has been answered multiple times) and make sure to follow Amazon's guide on setting up IAM users and roles.

**Reply by makaloni6** [reply]
Thank you for answer.

I finally resolve this problem.
